src/main.rs

```diff
@@ -1,32 +1,40 @@
-use std::env;
-use std::fs;
+#![feature(seek_stream_len)]
+#![warn(clippy::pedantic)]
+#![deny(missing_docs, missing_debug_implementations, clippy::all)]
```

Comment:
these are more compiler flags

- `deny(missing_docs)` - this will prevent your code from compiling if you have failed to document anything which is in your public API
- `deny(missing_debug_implementations` - prevent code from compiling if any struct is missing a `Debug` implementation (this prevents errors involving this type being printed nicely to console)
- `deny(clippy::all)` - prevent compiling for any standard clippy lint. all the lints in this category are very sensible, so this is a good one to enforce